src/config.ts

```diff
@@ -0,0 +1,103 @@
+  get: (target, key: string) => {
+    return (typeof target[key] === 'object' && target[key] !== null)
+      ? new Proxy(target[key], makeHandler(proxyConfig, path.concat(key)))
+      : target[key]
+  },
+
+  set(target, key: string, value) {
+    if ((Array.isArray(target) && key === 'length') || JSON.stringify(target[key]) === JSON.stringify(value)) {
+      return Reflect.set(target, key, value)
+    }
+
+    let result = value;
+    (Array.isArray(target) ? path : path.concat(key)).reverse().forEach((k, idx) => {
+      result = { [k]: Array.isArray(target) && idx === 0 ? [result] : result }
+    })
+
+    // @ts-ignore
+    Object.entries(configMerger(proxyConfig, result)).forEach(([k, v]) => proxyConfig[k] = v)
+    return Reflect.set(target, key, value)
+  }
+})
+
+/**
+ * Loads all possible Tailwind CSS configuration for a Nuxt project.
+ *
+ * @param moduleOptions configuration for the module
+ * @param nuxt nuxt app
+ * 
+ * @returns template with all configs handled
+ */
+export default async function loadTwConfig(moduleOptions: ModuleOptions, nuxt = useNuxt()) {
+  const [configPaths, contentPaths] = await resolveModulePaths(moduleOptions.configPath, nuxt)
+  const configProxies: [string, PartialTWConfig][] = configPaths.map((configPath) => [configPath, {}])
+
+  const tailwindConfig = await Promise.all((
+    configPaths.map(async (configPath, idx, paths) => {
+      let _tailwindConfig: PartialTWConfig | undefined
+
+      try {
+        _tailwindConfig = loadConfig(configPath)
+      } catch (e) {
+        logger.warn(`Failed to load Tailwind config at: \`./${relative(nuxt.options.rootDir, configPath)}\``, e)
+      }
+
+      // Transform purge option from Array to object with { content }
+      if (_tailwindConfig && !_tailwindConfig.content) {
+        configProxies[idx][1].content = _tailwindConfig.purge
+      }
+
+      await nuxt.callHook('tailwindcss:loadConfig', _tailwindConfig && new Proxy(_tailwindConfig, makeHandler(configProxies[idx][1])), configPath, idx, paths)
+      return _tailwindConfig || {}
+    }))
+  ).then((configs) => configs.reduce(
+    (prev, curr) => configMerger(curr, prev),
+    // internal default tailwind config
+    configMerger(moduleOptions.config, { content: contentPaths })
+  )) as TWConfig
+
+  // Allow extending tailwindcss config by other modules
+  const configProxy: PartialTWConfig = {}
+  await nuxt.callHook('tailwindcss:config', new Proxy(tailwindConfig, makeHandler(configProxy)))
+
+  const mergerTemplate = addTemplate({ src: createResolver(import.meta.url).resolve('./runtime/merger.mjs'), write: true })
+
+  return {
+    template: addTemplate({
+      filename: 'tailwind.config.cjs',
+      write: true,
+      getContents: () => {
+        return [
+          `const { configMerger } = require(${JSON.stringify(mergerTemplate.dst)});`,
+          configPaths.map((p, idx) => `const cfg${idx} = require(${JSON.stringify(relative(nuxt.options.buildDir, p))});`).join('\n'),
```

Comment:
tailwind will not consider it a tracked dependency if path is not relative

Reply:
need to exclude node_modules from these. moreover, if dir is same, it won't give us "./" prefix and it won't resolve.